arch/x86/kernel/slaunch.c

```diff
@@ -27,13 +27,72 @@
 #include <asm/e820/api.h>
 #include <asm/bootparam.h>
 #include <asm/setup.h>
+#include <linux/fs.h>
 #include <linux/slaunch.h>
 
+#define MSG_BUFFER_LEN 20
+#define MSG_ERROR "Mapping Error\n"
+
 #define PREFIX	"SLAUNCH: "
 
+#define DECLARE_PUB_READ(reg_name, reg_offset, reg_size)					\
+static ssize_t reg_name##_read(struct file *flip, char *buffer, size_t len, loff_t *offset) {	\
+	txt_info_to_buffer(reg_offset, reg_size);						\
+	return simple_read_from_buffer(buffer, len, offset, &msg_buffer, MSG_BUFFER_LEN);	\
+}												\
+static const struct file_operations reg_name##_ops = {						\
+	.read = reg_name##_read,								\
+	.write = reg_write									\
+};
+
+
+static char msg_buffer[MSG_BUFFER_LEN];
+
 static u32 sl_flags;
 static struct sl_ap_wake_info ap_wake_info;
 
+static ssize_t reg_write(struct file *file, const char __user *buf, size_t datalen, loff_t *ppos) {
+	return -EINVAL;
+}
+
+static void txt_info_to_buffer(unsigned int offset, int size) {
```

Comment:
You are passing the result of `sizeof` as the size param and `sizeof` is of type size_t which is unsigned thus making an implicit cast to a signed int. This then results in the comparison between a signed int and unsigned int in your switch statement later.